### PR TITLE
Feature/data table functionality

### PR DESCRIPTION
**Pull request by 000-ooo-000**
- Implements handler to allow the table to reload from an external event.
- Adds new story with sample code.
- Adds `style` and `className` props to table and table actions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [2.0.4] - 2022-04-22
```
Comment (neotrow) [inline CHANGELOG.md:10]: @000-ooo-000 

You can ommit the this line. Just add your changes under the unreleased sections. Will automatically be copied over by the release pipeline :-)

File: src/lib/DataTable/DataTableRouted.tsx [hunk 2]
```diff
@@ -35,6 +39,11 @@ export function DataTableRouted<T, TFilter, TRouteNames>({
   const [orderState, setOrderState] = useState<OrderOption>({ orderBy: columns[0].dataField, asc: true });
   const filterRefs = useRef<Filters>({});
 
+  if (handlers)
```
Comment (neotrow) [inline src/lib/DataTable/DataTableRouted.tsx:42]: Correct me if im wrong but this is executed on every render now. I think doing it once would be sufficient? A useEffect with an empty dependecy array should do the trick.

File: CHANGELOG.md (hunk 1 above)
Comment (000-ooo-000) [reply CHANGELOG.md]: Ommitted line

File: src/lib/DataTable/DataTableRouted.tsx (hunk 2 above)
Comment (000-ooo-000) [reply src/lib/DataTable/DataTableRouted.tsx]: You're right, did not think about that. Added `useEffect`

Comment (neotrow) [reply src/lib/DataTable/DataTableRouted.tsx]: Did you test it with the new code and when you are for example on page 2 or already have a filter set? 
I'm afraid that closures might take a snapshot of the variables and therefor misbehave and always pass the initial values to the `loadPage` function. But i'm not 100% sure.

File: src/lib/DataTable/DataTableRouted.tsx [hunk 3]
```diff
@@ -116,14 +125,18 @@ export function DataTableRouted<T, TFilter, TRouteNames>({
 
   return (
     <React.Fragment>
-      <Table striped hover size="sm">
+      <Table striped hover size="sm" className={className} style={style}>
```
[the remark this answers is not in the document]
Comment (neotrow) [reply src/lib/DataTable/DataTableRouted.tsx]: thx!

File: src/lib/DataTable/DataTableRouted.tsx (hunk 2 above)
Comment (000-ooo-000) [reply src/lib/DataTable/DataTableRouted.tsx]: Unfortunately it did snapshot all the arguments, therefore I removed `useEffect`.

Comment (neotrow) [reply src/lib/DataTable/DataTableRouted.tsx]: I would highly suggest to write a jest test for this. (Checking if the supplied arguments to the load page are correct)

Still unsure if there is not a better option than doing it on every render. Seems a bit overkill but maybe i'm wrong. 
IIRC refs would work. But than you would have to keep a state and a ref which also seems a bit stupid...

**Review by neotrow — CHANGES_REQUESTED** [review]
As discussed, please write some jest tests for it. I suggest the following:

snapshot: check that the css propertiers are applied to the table when supplied
snapshot: check that the css propertiers are applied to the table actions when supplied
check that the reload function behaves correctly. (Use jest.fn) and afterwards check the arguments and number of calls. Especially after you switched the page and changed the filters. To make sure the supplied arguments are up to date.

File: src/lib/DataTable/DataTableInterfaces.ts [hunk 4]
```diff
@@ -1,9 +1,12 @@
+/* eslint-disable max-lines */
```
Comment (neotob) [inline src/lib/DataTable/DataTableInterfaces.ts:1]: can you may put something like this? like this we don't have it completely open and need to reconsider again if we wanna split it if we add more code. 

/* eslint max-lines: ["error", 220] */ // comment why :D

Comment (neotob) [inline src/lib/DataTable/DataTableInterfaces.ts:1]: can you may put something like this? like this we don't have it completely open and need to reconsider again if we wanna split it if we add more code. 

/* eslint max-lines: ["error", 220] */ // comment why :D

File: src/lib/DataTable/DataTableRouted.tsx (hunk 2 above)
Comment (000-ooo-000) [reply src/lib/DataTable/DataTableRouted.tsx]: Updated snapshots and added reloading tests

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+-   Dynamic table handler to reload the data from an external source.
+-   Added new story to reload data with sample code.
+-   Implemented className and style properties to the table and the actions.
+-   Added reload tests.
+-   Changed max-lines allowed in DataTableInterfaces.ts.
```
Comment (neotrow) [inline CHANGELOG.md:14]: This really is a detail but this should go under a `Changed` section.

ref: https://keepachangelog.com/en/1.0.0/

File: src/lib/DataTable/DataTableRouted.tsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable max-lines */
```
Comment (neotrow) [inline src/lib/DataTable/DataTableRouted.tsx:1]: @000-ooo-000 

could you add something similar here to allow 230. 

Thanks

File: test/__mocks__/DataTable/dynamic.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { useRef } from "react";
```
Comment (neotrow) [inline test/__mocks__/DataTable/dynamic.tsx:1]: @000-ooo-000 

Cool stuff, didn't know that this is possible 🎉

**Review by neotrow — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (000-ooo-000) [reply CHANGELOG.md:16]: Moved changes to the correct section

File: src/lib/DataTable/DataTableRouted.tsx (hunk 6 above)
Comment (000-ooo-000) [reply src/lib/DataTable/DataTableRouted.tsx]: Modified rule

**Review by neotrow — APPROVED** [review]

**Pull request merged**